### PR TITLE
Close #12438: Refactor RIDE_COMPONENT_TYPE to use strong enum

### LINKED ISSUE #12438: Refactor RIDE_COMPONENT_TYPE to use strong enum

### PR DESCRIPTION
**Pull request by Balvald**
Refactored RideComponentType

### CODE REVIEW

File: src/openrct2-ui/interface/ViewportInteraction.cpp [hunk 1]
```diff
@@ -363,7 +363,8 @@ InteractionInfo ViewportInteractionGetItemRight(const ScreenCoordsXY& screenCoor
             }
 
             ride->FormatNameTo(ft);
-            ft.Add<rct_string_id>(RideComponentNames[RideTypeDescriptors[ride->type].NameConvention.station].capitalised);
+            ft.Add<rct_string_id>(
+                RideComponentNames[EnumValue(RideTypeDescriptors[ride->type].NameConvention.station)].capitalised);
```
Comment (duncanspumpkin) [inline src/openrct2-ui/interface/ViewportInteraction.cpp:367]: Make an accessor function for this as it is used in tons of places.

`constexpr const RideComponentName& GetRideComponentName(const RideComponentType type){ return RideComponentNames[EnumValue(type)];}`

File: src/openrct2/ride/RideData.cpp [hunk 2]
```diff
@@ -312,6 +312,11 @@ constexpr const RideTypeDescriptor RideTypeDescriptors[RIDE_TYPE_COUNT] = {
     /* RIDE_TYPE_HYBRID_COASTER                     */ HybridCoasterRTD,
 };
 
+constexpr const RideComponentName& GetRideComponentName(const RideComponentType type)
+{
+    return RideComponentNames[EnumValue(type)];
```
Comment (duncanspumpkin) [inline src/openrct2/ride/RideData.cpp:317]: constexpr functions go in the header file.

Comment (Balvald) [reply src/openrct2/ride/RideData.cpp]: Is noted. thanks.

**Review by duncanspumpkin — APPROVED** [review]

**Review by tupaschoal — APPROVED** [review]

**Comment by tupaschoal** [comment]
Thanks @Balvald. One note for next time: when you solve issues/comments pointed out by other people, make sure you mark them as resolved.

**Pull request merged**